code/modules/paperwork/desk_bell.dm

```diff
@@ -57,12 +57,13 @@
 
 /obj/item/desk_bell/proc/on_signal()
 	SIGNAL_HANDLER  // COMSIG_ASSEMBLY_PULSED
-	INVOKE_ASYNC(src, PROC_REF(try_ring))
+	INVOKE_ASYNC(src, PROC_REF(try_ring), null, TRUE) // user=null, from_signaler=TRUE (if nobody broke it)
```

Comment:
Why comment this? Just set the keyword args directly in the proc call

Reply:
How do do you mean "set the keyword args directly in the proc call"? I have not found a way to pass arguments by name through INVOKE_ASYNC. When I pass an associative list, that list ends up as the first positional argument to the called proc. I also cannont pass them as keyword arguments to INVOKE_ASYNC (aka the underlying ImmediateInvokeAsync), since that takes set keyword and an arbitrary number of positional arguments, but not an arbitrary number of keyword arguments. Passing in additional keyword arguments results in dreamchecker error and a runtime error.
I'd love to be wrong about it though.

Reply:
Oh, sorry. I was on mobile and missed that it was being done through invoke async. That's on me